rust/main/chains/hyperlane-sealevel/src/provider.rs

```diff
@@ -92,41 +85,6 @@ impl SealevelProvider {
         Ok(sender)
     }
 
-    fn recipient(hash: &H512, txn: &UiTransaction) -> ChainResult<H256> {
-        let message = Self::parsed_message(txn)?;
-
-        let programs = message
-            .instructions
-            .iter()
-            .filter_map(|ii| {
-                if let UiInstruction::Parsed(iii) = ii {
-                    Some(iii)
-                } else {
-                    None
-                }
-            })
-            .map(|ii| match ii {
-                UiParsedInstruction::Parsed(iii) => &iii.program_id,
-                UiParsedInstruction::PartiallyDecoded(iii) => &iii.program_id,
-            })
-            .filter(|program_id| !NATIVE_PROGRAMS.contains(*program_id))
-            .collect::<Vec<&String>>();
-
-        if programs.len() > 1 {
```

Comment:
tbf I could also see it being valid for us to just have the heuristic be "pick whatever the first non native program is, regardless of whether it's the instruction responsible for the message or IGP payment"